cmd/dp-filter-api/main.go

```diff
@@ -51,12 +60,13 @@ func main() {
 
 	client := rchttp.DefaultClient
 	datasetAPI := dataset.NewDatasetAPI(client, cfg.DatasetAPIURL, cfg.DatasetAPIAuthToken)
-
+	observationStore := observation.NewStore(neo4jDriver)
```

Comment:
Should the API / PreviewDatasetStore maintain a connection pool to Neo4j? It looks like its using a single connection ATM

Reply:
Done, I made another PR to address this https://github.com/ONSdigital/dp-filter/pull/3